diffr/data_model.py

```diff
@@ -119,22 +119,26 @@ class Diff(object):
     It can also can be wrapped in a DiffItem as an item of a higher level Diff
     ie the objects being diffed are nested.
 
-    :attribute type: The type of the objects being diffed
-    :attribute diffs: A list containing all of the DiffItems including an
-        unchanged ones.
-    :attribute depth: Indicates how deep this diff is in a nested diff.
-
-    Diffs are uniquely identified by the values of their attributes.
+    :property type: The type of the objects being diffed
+    :property depth: Indicates how deep this diff is in a nested diff.
     '''
     def __init__(self, obj_type, diffs, depth=0):
-        self.type = obj_type
-        self.diffs = tuple(diffs)
+        self._type = obj_type
+        self._diffs = tuple(diffs)
         # flag used by __format__ and the DiffContext context manager
-        self.context_limit = None
-        self.depth = depth
-        self._indent = '   ' * depth
-        self.start = unchanged('{}('.format(self.type))
```

Comment:
use self.type.__name__ to print just the constructor. "It reduces symbol clutter" Olivier